interfaces/builtin/udisks2.go

```diff
@@ -0,0 +1,423 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+
+	"github.com/snapcore/snapd/interfaces"
+)
+
+const udisks2PermanentSlotAppArmor = `
+# Description: Allow operating as the udisks2. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# DBus accesses
+#include <abstractions/dbus-strict>
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.freedesktop.UDisks2",
+
+# Allow unconfined to talk to us
+dbus (receive, send)
+    bus=system
+    path=/org/freedesktop/UDisks2{,/**}
+    interface=org.freedesktop.DBus*
+    peer=(label=unconfined),
+
+# Needed for mount/unmount operations
+capability sys_admin,
+
+# Allow reading existing mount points
+@{PROC}/@{pid}/mountinfo r,
+@{PROC}/swaps r,
+
+# Allow scanning of devices
+network netlink raw,
+/run/udev/data/b[0-9]*:[0-9]* r,
+/sys/devices/**/block/** r,
+
+# Mount points could be in /run/media/<user>/* or /media/<user>/*
+/run/systemd/seats/* r,
+/{,run/}media/{,**} rw,
+mount /dev/{sd*,mmcblk*} -> /{,run/}media/**,
```

Comment:
It will use whatever the defaults are for `mount` on that system. They can be overridden, but only these options are allowed: `exec`, `noexec`, `nodev`, `nosuid`, `atime`, `noatime`, `nodiratime`, `ro`, `rw`, `sync`, `dirsync`